features/createroom/impl/src/main/kotlin/io/element/android/features/createroom/impl/configureroom/ConfigureRoomView.kt

```diff
@@ -0,0 +1,375 @@
+        )
+    }
+}
+
+@Composable
+fun Avatar(
+    avatarUri: Uri?,
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit = {},
+) {
+    val commonModifier = modifier
+        .size(70.dp)
+        .clip(CircleShape)
+        .clickable(onClick = onClick)
+
+    if (avatarUri != null) {
+        val context = LocalContext.current
+        val model = ImageRequest.Builder(context)
+            .data(avatarUri)
+            .build()
+        AsyncImage(
+            modifier = commonModifier,
+            model = model,
+            contentDescription = null,
+        )
+    } else {
+        Box(modifier = commonModifier.background(LocalColors.current.quinary)) {
+            Icon(
+                imageVector = Icons.Outlined.AddAPhoto,
+                contentDescription = "",
+                modifier = Modifier
+                    .align(Alignment.Center)
+                    .size(40.dp),
+                tint = MaterialTheme.colorScheme.secondary,
+            )
+        }
+    }
+}
+
+@Composable
+fun RoomTopic(
+    topic: String,
+    modifier: Modifier = Modifier,
+    onTopicChanged: (String) -> Unit = {},
+) {
+    LabelledTextField(
+        modifier = modifier,
+        label = stringResource(R.string.screen_create_room_topic_label),
+        value = topic,
+        placeholder = stringResource(R.string.screen_create_room_topic_placeholder),
+        onValueChange = onTopicChanged,
+        maxLines = 3,
+    )
+}
+
+@Composable
+fun RoomPrivacyOptions(
+    selected: RoomPrivacy?,
+    modifier: Modifier = Modifier,
+    onOptionSelected: (RoomPrivacy) -> Unit = {},
+) {
+
+    data class RoomPrivacyItem(
+        val privacy: RoomPrivacy,
+        val icon: ImageVector,
+        val title: String,
+        val description: String,
+    )
+
+    val items = RoomPrivacy.values().map {
+        when (it) {
+            RoomPrivacy.Public -> RoomPrivacyItem(
+                privacy = it,
+                icon = Icons.Outlined.Lock,
+                title = stringResource(R.string.screen_create_room_private_option_title),
+                description = stringResource(R.string.screen_create_room_private_option_description),
+            )
+            RoomPrivacy.Private -> RoomPrivacyItem(
+                privacy = it,
+                icon = Icons.Outlined.Public,
+                title = stringResource(R.string.screen_create_room_public_option_title),
+                description = stringResource(R.string.screen_create_room_public_option_description),
+            )
+        }
+    }
+    Column(modifier = modifier.selectableGroup()) {
+        items.forEach { item ->
+            RoomPrivacyOption(
+                privacy = RoomPrivacy.Private,
+                icon = item.icon,
+                title = item.title,
+                description = item.description,
+                isSelected = selected == item.privacy,
+                onOptionSelected = { onOptionSelected(item.privacy) }
```

Comment:
Ha the bug is not visible since you are not using the value here, like this:

```suggestion
                onOptionSelected = { onOptionSelected(it) }
```